src/iOS/AppDelegate.cs

```diff
@@ -201,8 +201,8 @@ public override void OnResignActivation(UIApplication uiApplication)
 
         public override void DidEnterBackground(UIApplication uiApplication)
         {
-            _storageService.SaveAsync(Constants.LastActiveTimeKey, _deviceActionService.GetActiveTime());
-            _messagingService.Send("slept");
+            _storageService?.SaveAsync(Constants.LastActiveTimeKey, _deviceActionService.GetActiveTime());
+            _messagingService?.Send("slept");
```

Comment:
`DidEnterBackground` only runs for roughly ~5s, we might have to run this in a background task.

https://docs.microsoft.com/en-us/xamarin/ios/app-fundamentals/backgrounding/ios-backgrounding-techniques/ios-backgrounding-with-tasks

Reply:
Those methods don't take much time AFAIK, but yes it would be a great idea to run them in a background task